### ISSUE TITLE
How to addVariable under global namespace?

### ISSUE THREAD

**Issue by hcaihao** [vinniefalco/LuaBridge#143]
Runtime error when addVariable under global
```
	luabridge::getGlobalNamespace(L)
		.addVariable("AppPath", &g_appPath, false)
		;
```

but addFunction under global is ok
```
	luabridge::getGlobalNamespace(L)
		.addFunction("IsFileExist", IsFileExist)
		;
```

**Comment by theDagda** [comment]
this was posted a while ago, but if you still couldn't figure out, you need to change it like so
	luabridge::getGlobalNamespace(L)
		.addVariable<Type>("AppPath", &g_appPath, false)
		;
`
Where type is whatever g_appPath's type is

**Comment by theDagda** [comment]
```c++	
luabridge::getGlobalNamespace(L)
		.addVariable<type>("AppPath", &g_appPath, false)
		;
```
where <type> is whatever type your &g_appPath is.

**Comment by dmitry-t** [comment]
TL;DR Unfortunately, adding a variable or property to the global namespace is not possible.
So I'll just make the LuaBridge throwing an exception in such cases.

Adding a variable means Lua gets a userdata holding a pointer to the C++ value. Then, in order to get/set the userdata in Lua, the userdata parent table must have a metatable with `__index` and `__newindex` functions (otherwise that userdata would be just replaced i.e. thrown away from Lua). The problem is that it's not possible to simply set a metatable to the global table (`_G`): this table is [not used by Lua](https://www.lua.org/manual/5.1/manual.html#5.1); to attach a metatable to `_G` you need to setup a new environment ([lua_setfenv](https://www.lua.org/manual/5.1/manual.html#lua_setfenv)). And in Lua 5.2 the function setting an environment [is removed from API](https://www.lua.org/manual/5.2/manual.html#8.3) at all.

**Comment by dmitry-t** [comment]
Exceptions added in #167.
